Proposal: Update README for use `sdk` and `sdk-mock` more properly (#76)
DeployGate/deploygate-android-sdk: README.md

}

dependencies {

>     // use full implementation for debug builds

Reviewer: ```suggestion
    // Use SDK dependency for variants like debug.
```

    debugImplementation 'com.deploygate:sdk:<latest version>'
    

>     // use mocked implementation for release builds

Reviewer: ```suggestion
    // Use no-op implementation for variants you would like to disable DeployGate SDK.
```

Author: Sound good to me.
The scene of need to disable DeployGate SDK maybe not only release build 💡

</application>
```

> And you also need to add `DeployGate.install()` to your implementation.

Reviewer: ```suggestion
And also, you need to call `DeployGate#install` in your Application class, ContentProvider or AndroidX Startup Initializer.
```

Author: Sorry, I have misunderstood.
SDK is didn't supported  to call method with named arguments.

For example, add to your custom application class, content provider, or else.

```java
DeployGate.install(context, /** forceApplyOnReleaseBuild */ false);
```

> ```kotlin
> DeployGate.install(app = context, forceApplyOnReleaseBuild = false)
> ```

Reviewer: ```suggestion
```

No, we are not providing Kotlin-friendly syntax like named arguments because SDK is built on Java.

Author: Sorry, I have misunderstood.
SDK is didn't supported  to call method with named arguments.

> You may want to remove DeployGate SDK and related code in production build
> to reduce your app's footprint.
> 
> In addition, you may want to disable all features provided by DeployGate SDK
> on artifacts for production release.
> 
> For your convenience, we provide "Mock" SDK that replaces every function call
> to empty implementation so you don't have to modify your code to switch the builds.

Reviewer: ```suggestion
Do you want to disable DeployGate SDK on production builds? If so, please use `sdk-mock` dependency for production builds instead of `sdk`. `sdk-mock` dependency has public interfaces that are same as of `sdk` but their implementations are empty, so you don't have to modify your app code for specific build variants.
```
